From 39ce26d8bc40ec2b2929ac7d612ccead6d96e64a Mon Sep 17 00:00:00 2001
From: Ovidiu Panait <ovidiu.panait@windriver.com>
Date: Fri, 31 Aug 2018 07:27:08 +0000
Subject: [PATCH] dmaengine: Fix softip dma test client compilation

Fix the following compile errors when CONFIG_XILINX_DMATEST=y:
error: implicit declaration of function 'put_task_struct'; did you mean 'get_task_cred'? [-Werror=implicit-function-declaration]
| put_task_struct(thread->task);
| ^~~~~~~~~~~~~~~
...
error: implicit declaration of function 'get_task_struct'; did you mean 'get_task_cred'? [-Werror=implicit-function-declaration]
| get_task_struct(thread->task);
| ^~~~~~~~~~~~~~~

This is an upstream issue that was fixed as part of the following
linux-xlnx merge-commit:

commit 818f168696f561c127f161379eb5b8d1835218a2
Merge: 9c2e29b2c81d bebc6082da0a
Author: Michal Simek <michal.simek@xilinx.com>
Date:   Thu Jan 11 13:58:55 2018 +0100

    Merge tag 'v4.14' into master

    Linux 4.14
    Changes:
...
	dma:
	- dmaengine: Fix softip dma test client compilation (by Kedar)
...

Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 drivers/dma/xilinx/axidmatest.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/dma/xilinx/axidmatest.c b/drivers/dma/xilinx/axidmatest.c
index 831d9bd..fce1561 100644
--- a/drivers/dma/xilinx/axidmatest.c
+++ b/drivers/dma/xilinx/axidmatest.c
@@ -19,6 +19,7 @@
 #include <linux/random.h>
 #include <linux/slab.h>
 #include <linux/wait.h>
+#include <linux/sched/task.h>
 #include <linux/dma/xilinx_dma.h>
 
 static unsigned int test_buf_size = 16384;
-- 
1.7.9.5

